tests/test_mongodb_incremental_open_transactions.py

```diff
@@ -0,0 +1,298 @@
+            exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+            menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+            # verify the tap discovered the right streams
+            found_catalogs = menagerie.get_catalogs(conn_id)
+
+            # assert we find the correct streams which includes all collections which are part of session1 and session2
+            self.assertEqual(self.expected_check_streams_sync_1(),
+                             {c['tap_stream_id'] for c in found_catalogs})
+
+            # Select streams and add replication method metadata
+            for stream_catalog in found_catalogs:
+                annotated_schema = menagerie.get_annotated_schema(conn_id, stream_catalog['stream_id'])
+                additional_md = [{ "breadcrumb" : [], "metadata" : {'replication-method' : 'INCREMENTAL', 'replication_key': 'int_field'}}]
+                selected_metadata = connections.select_catalog_and_fields_via_metadata(conn_id,
+                                                                                        stream_catalog,
+                                                                                        annotated_schema,
+                                                                                        additional_md)
+
+            # run full table sync
+            sync_1 = runner.run_sync_mode(self, conn_id)
+
+            # check exit status
+            exit_status = menagerie.get_exit_status(conn_id, sync_1)
+            menagerie.verify_sync_exit_status(self, exit_status, sync_1)
+
+            # streams that we synced are the ones that we expect to see
+            records_by_stream = runner.get_records_from_target_output()
+            record_count_by_stream = runner.examine_target_output_file(self,
+                                                                       conn_id,
+                                                                       self.expected_sync_streams_1(),
+                                                                       self.expected_pks_1())
+
+            # validate the record count in collections which are part of session1 and session2, should not read updates on coll 1 and coll 2 and insert on coll 3. Because the transaction is not committed
+            self.assertEqual(self.expected_row_counts_sync_1(), record_count_by_stream)
+
+            # validate there are no duplicates replicated as part of sync1
+            records = {}
+            pk_dict = {}
+            for stream in self.expected_sync_streams_1():
+                records[stream] = [x for x in records_by_stream[stream]['messages'] if x.get('action') == 'upsert']
+                pk = set()
+                for i in range(len(records[stream])):
+                    pk.add(records[stream][i]['data']['int_field'])
+                pk_dict[stream] = len(pk)
+
+            self.assertEqual(self.expected_row_counts_sync_1(), pk_dict)
```

Comment:
This is a little tough for me to follow. I think it could be simplified a bit. But I won't block on it since  isn't critical to the functionality of the test, just the readability.